dart/lib/src/sentry_client.dart

```diff
@@ -52,14 +52,16 @@ class SentryClient {
     dynamic stackTrace,
     dynamic hint,
   }) async {
-    event =
-        await _processEvent(event, eventProcessors: _options.eventProcessors);
-
-    // dropped by sampling or event processors
-    if (event == null) {
+    if (_sampleRate()) {
```

Comment:
these changes are to follow our event pipeline and also because when event processors run, the event should be already prepared.